source/Octopus.Tentacle.Tests.Integration/Support/RunningTentacle.cs

```diff
@@ -93,14 +94,22 @@ public async Task Restart(CancellationToken cancellationToken)
 
         public async ValueTask DisposeAsync()
         {
+            using var disposeCancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(10));
```

Comment:
I suspect that something in this Dispose is the culprit so I added a 10 second timeout to anything that already accepted a cancellation token.